File: OOP/02_base/04_incapsulation.py
class Student:
    def __init__(self, name, last_name):
        self.__name = name
        self.__last_name = last_name

    def get_name(self):
        return self.__name

    def set_name(self, name):
        self.__name = name;

    def get_last_name(self):
        return self.__last_name

    def set_last_name(self, last_name):
        self.__last_name = last_name;

    def get_full_name(self):
        print("Полное имя: " + self.__name + ' ' + self.__last_name)


student1 = Student('Сергей', 'Иванов')
student1.get_full_name()

print(student1.get_name())
student1.set_name('Петр')
print(student1.get_name())

# print(student1.__name)
# print(student1._Student__name)

print(dir(student1))

File: OOP/02_base/04_incapsulation.php
<?php
namespace Popov\Lection2\Ex4;

class Student
{
    private string $name;
    private string $lastName;

    public function __construct(string $name, string $lastName)
    {
        $this->name = $name;
        $this->lastName = $lastName;
    }

    public function getName(): string
    {
        return $this->name;
    }

    public function setName(string $name): void
    {
        $this->name = $name;
    }

    public function getLastName(): string
    {
        return $this->lastName;
    }

    public function setLastName(string $lastName)
    {
        $this->lastName = $lastName;
    }

    public function showFullName()
    {
        echo "Полное имя: ".$this->name." ".$this->lastName.PHP_EOL;
    }
}


$student1 = new Student("Сергей", "Иванов");

echo $student1->getName().PHP_EOL;
//echo $student1->name;

$student1->showFullName();
$student1->setName("Николай");
$student1->showFullName();

//
////echo $student1->name;
//$student1->age = 24;
//echo $student1->age.PHP_EOL;

// PHP Fatal error:  Uncaught Error: Cannot access private property
